platforms/qurt/cmake/qurt_reqs.cmake

```diff
@@ -0,0 +1,222 @@
+		set(${out} "${${out}} ${item}")
+	endforeach()
+endmacro(list2string)
+
+set(HEXAGON_START_LINK_FLAGS)
+list2string(HEXAGON_START_LINK_FLAGS
+	${HEXAGON_ARCH_FLAGS}
+	-shared
+	-call_shared
+	-G0
+	${TOOLSLIB}/initS.o
+	"-o <TARGET>"
+	-L${TOOLSLIB}
+	-Bsymbolic
+	${TOOLSLIB}/libgcc.a
+	--wrap=malloc
+	--wrap=calloc
+	--wrap=free
+	--wrap=realloc
+	--wrap=memalign
+	--wrap=__stack_chk_fail
+	-lc
+	"-soname=<TARGET_SONAME>"
+	)
+
+set(HEXAGON_END_LINK_FLAGS)
+list2string(HEXAGON_END_LINK_FLAGS
+	--start-group
+	-lgcc
+	--end-group
+	${TOOLSLIB}/finiS.o
+	)
+
+set(CMAKE_C_CREATE_SHARED_LIBRARY
+	"${HEXAGON_LINK} ${HEXAGON_START_LINK_FLAGS} --start-group --whole-archive <OBJECTS> <LINK_LIBRARIES> --end-group ${HEXAGON_END_LINK_FLAGS}")
+
+set(CMAKE_CXX_CREATE_SHARED_LIBRARY
+	"${HEXAGON_LINK} ${HEXAGON_START_LINK_FLAGS} --start-group ${TOOLSLIB}/libc++.so.1 ${TOOLSLIB}/libc++abi.so.1 ${HEXAGON_SDK_ROOT}/libs/weak_refs/ship/hexagon_toolv84/weak_refs.so --end-group --start-group --whole-archive <OBJECTS> <LINK_LIBRARIES> --end-group ${HEXAGON_END_LINK_FLAGS}")
+
+set(DYNAMIC_LIBS -Wl,${TOOLSLIB}/libc++.a)
+
+# Base CPU flags for each of the supported architectures.
+set(ARCHCPUFLAGS
+	-m${V_ARCH}
+	-G0
+	)
+
+add_definitions(
+	-D __QURT
+	-D _PROVIDE_POSIX_TIME_DECLS
+	-D _TIMER_T
+	-D _HAS_C9X
+	-D restrict=__restrict__
+	-D noreturn_function=
+	)
+
+# Language-specific flags
+set(ARCHCFLAGS
+	-D__CUSTOM_FILE_IO__
+	)
+
+set(ARCHCXXFLAGS
+	-DCONFIG_WCHAR_BUILTIN
+	-D__CUSTOM_FILE_IO__
+	)
+
+exec_program(${CMAKE_CXX_COMPILER} ${CMAKE_CURRENT_SOURCE_DIR} ARGS -print-libgcc-file-name OUTPUT_VARIABLE LIBGCC)
+exec_program(${CMAKE_CXX_COMPILER} ${CMAKE_CURRENT_SOURCE_DIR} ARGS -print-file-name=libm.a OUTPUT_VARIABLE LIBM)
+set(EXTRA_LIBS ${EXTRA_LIBS} ${LIBM})
+
+# Full optimization and Link Time Optimization (LTO)
+set(QURT_OPTIMIZATION_FLAGS -O3 -flto)
+
+# Flags we pass to the C compiler
+list2string(CFLAGS
+	${ARCHCFLAGS}
+	${ARCHCPUFLAGS}
+	${QURT_OPTIMIZATION_FLAGS}
+	${HEXAGON_INCLUDE_DIRS}
+	)
+
+# Flags we pass to the C++ compiler
+list2string(CXXFLAGS
+    -Wno-inconsistent-missing-override
+	${ARCHCXXFLAGS}
+	${ARCHCPUFLAGS}
+	${QURT_OPTIMIZATION_FLAGS}
+	${HEXAGON_INCLUDE_DIRS}
+	)
+
+# Set cmake flags
+list2string(QURT_CMAKE_C_FLAGS
+	${CMAKE_C_FLAGS}
+	${CFLAGS}
+	)
+
+set(CMAKE_C_FLAGS "${QURT_CMAKE_C_FLAGS}")
+
+list2string(QURT_CMAKE_CXX_FLAGS
+	${CMAKE_CXX_FLAGS}
+	${CXXFLAGS}
+	)
+
+set(CMAKE_CXX_FLAGS ${QURT_CMAKE_CXX_FLAGS})
+
+# Flags we pass to the linker
+# CMake make test builds of apps to validate the compiler
+# These settings enable CMake to build the required test apps
+list2string(CMAKE_EXE_LINKER_FLAGS
+	-m${V_ARCH}
+	-mG0lib
+	-G0
+	-fpic
+	-shared
+	-Wl,-Bsymbolic
+	-Wl,--wrap=malloc
+	-Wl,--wrap=calloc
+	-Wl,--wrap=free
+	-Wl,--wrap=realloc
+	-Wl,--wrap=memalign
+	-Wl,--wrap=__stack_chk_fail
+	${DYNAMIC_LIBS}
+	-lc
+	${EXTRALDFLAGS}
+	)
+
+include (CMakeParseArguments)
+
+# Process DSP files
+function (QURT_LIB)
```

Comment:
QURT_LIB seems like another one where we can just use regular cmake directly?

The deploy helper could be done generically for all libs (in the platform/qurt/ layer) or specifically for a few special targets?